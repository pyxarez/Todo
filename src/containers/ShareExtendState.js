import React, { Component } from 'react';

const ShareExtendState = (CategoryContainer) => class extends Component {
  state = { extended: false }

  handleExtendClick = () => {
    this.setState((prevState) => ({ extended: !prevState.extended }));
  }

  getChildrenCount = (children) => {
    return React.Children.toArray(children).length;
  }

  render() {
    return (
      <CategoryContainer
        {...this.props}
        {...this.state}
        onExtend={this.handleExtendClick}
        getChildrenCount={this.getChildrenCount}/>
    );
  }
}

export default ShareExtendState;
