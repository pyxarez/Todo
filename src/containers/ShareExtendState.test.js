import React, { Component } from 'react';
import { shallow } from 'enzyme';

import ShareExtendState from './ShareExtendState';

const TestComponent = () => (<div className="my-test-component"/>);
const ExtendedComponent = ShareExtendState(TestComponent);

const setup = () => {
  const wrapper = shallow(<ExtendedComponent/>);

  return {
    wrapper
  }
};

describe('Containers :: ShareExtendState', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should change state by calling handleExtendClick', () => {
    const { wrapper } = setup();

    wrapper.find(TestComponent).prop('onExtend')();
    expect(wrapper.state().extended).toBe(true);
  });

  it('should test returning value of getChildrenCount', () => {
    const { wrapper } = setup();

    const testChildren = <div className="i-am-children"></div>;
    expect(wrapper.instance().getChildrenCount(testChildren)).toBe(1);
  });
});