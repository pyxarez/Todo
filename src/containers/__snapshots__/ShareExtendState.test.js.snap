exports[`Containers :: ShareExtendState should render self and subcomponents 1`] = `
<TestComponent
  extended={false}
  getChildrenCount={[Function]}
  onExtend={[Function]} />
`;
